connectors/sources/salesforce.py

```
@@ -0,0 +1,1443 @@
+        for record in records:
+            created_by_list.append(record.get("CreatedBy"))
+
+        return created_by_list
+
+    def _format_list(self, unformatted):
+        return sorted(
+            set(filter(None, [str(x).strip() for x in unformatted if x is not None]))
+        )
+
+
+class SalesforceDataSource(BaseDataSource):
+    """Salesforce"""
+
+    name = "Salesforce"
+    service_type = "salesforce"
+
+    def __init__(self, configuration):
+        super().__init__(configuration=configuration)
+        self.base_url = BASE_URL.replace("<domain>", configuration["domain"])
+        self.salesforce_client = SalesforceClient(
+            configuration=configuration, base_url=self.base_url
+        )
+        self.doc_mapper = SalesforceDocMapper(self.base_url)
+
+    def _set_internal_logger(self):
+        self.salesforce_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        return {
+            "client_id": {
+                "label": "Client ID",
+                "order": 1,
+                "tooltip": "The client id for your OAuth2-enabled connected app. Also called 'consumer key'",
+                "type": "str",
+                "value": "",
+            },
+            "client_secret": {
+                "label": "Client Secret",
+                "order": 2,
+                "sensitive": True,
+                "tooltip": "The client secret for your OAuth2-enabled connected app. Also called 'consumer secret'",
+                "type": "str",
+                "value": "",
+            },
+            "domain": {
+                "label": "Domain",
+                "order": 3,
+                "sensitive": True,
+                "tooltip": "The domain for your Salesforce instance. If your Salesforce URL is 'foo.my.salesforce.com', the domain would be 'foo'.",
+                "type": "str",
+                "value": "",
+            },
+        }
+
+    async def validate_config(self):
```

Should we add some programatic validations here?  Probably good to check that the config fields are non-null. We should throw an Exception otherwise.